task/upload.go

```diff
@@ -62,20 +62,20 @@ func handleUploadVOD(p handleUploadVODParams) (*TaskHandlerOutput, error) {
 	)
 	switch step {
 	case "":
-		// TODO: Move this input decryption logic to catalyst
-		if uploadParams := tctx.Params.Upload; uploadParams != nil {
-			var err error
-			inUrl, err = decryptInputFile(tctx, inUrl, *uploadParams)
-			if err != nil {
-				return nil, fmt.Errorf("error decrypting input file: %w", err)
-			}
-		}
 		fallthrough
 	case "rateLimitBackoff":
 		outputLocations, err := p.getOutputLocations()
 		if err != nil {
 			return nil, err
 		}
+		var encryption *clients.EncryptionPayload
+		params := tctx.Task.Params.Upload
+
+		if params != nil && params.Encryption.EncryptedKey != "" {
+			encryption = &clients.EncryptionPayload{
+				EncryptedKey: params.Encryption.EncryptedKey,
+			}
+		}
```

Comment:
Can you move this to `handleUploadVODParams` as I suggested? This function should not need to know or care what is the task type of the caller. It was made exactly to abstract the common logic between all of them (upload or transcode-file). 

So rather of re-checking the same conditions from which its args are constructed, it should have its arguments well defined via its input struct arg.